src/taipy/core/_scheduler/_dispatcher/_dispatcher.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import threading
+from abc import abstractmethod
+from multiprocessing import Lock
+from typing import Any, List
+
+from taipy.config import Config
+from taipy.logger._taipy_logger import _TaipyLogger
+
+from ...common.alias import JobId
+from ...data._data_manager_factory import _DataManagerFactory
+from ...data.data_node import DataNode
+from ...exceptions.exceptions import DataNodeWritingError
+from ...job._job_manager_factory import _JobManagerFactory
+from ...job.job import Job
+from ...task.task import Task
+
+
+class _JobDispatcher(threading.Thread):
+    def __init__(self, scheduler):
+        threading.Thread.__init__(self)
+        self.daemon = True
+        self.scheduler = scheduler
+        self.lock = Lock()
+        self.__logger = _TaipyLogger._get_logger()
+
+    def run(self):
+        while True:
+            # if not self.scheduler.jobs_to_run.empty() and self._can_execute():
+            # self.__execute_jobs()
+            # else:
+            from time import sleep
+
+            print("going to sleep")
+            sleep(1)
+
```

Comment:
Instead of doing a sleep(1), we can eventually modify the __execute_jobs method to benefit from the fact that jobs_to_run.get() blocks until there is an item in the queue.

You can have a look at this [link](https://docs.python.org/3/library/queue.html#queue.Queue.get)

In any case, this is more than fine for a first version.

Reply:
that' great thanks!! Then I'd say we should rearrange the condition to `if self._can_execute() and job := self.scheduler.jobs_to_run.get(block=True):` to make sure the processes are available to execute a job and we won't have the issue where we continuously getting new jobs out of the queue and no available processes to run it.